mautic/mautic
# fix country asset helper

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [X]
| Tests included	| [X]
| Issue(s) addressed                     | Fixes **it**

#### Description:

flag asset helper does attempt to get flag for country code while assuming it is country name. it won't find the file. secondly if file not found the boolean is returned instead of string which causes exception.

please check [comments](https://github.com/mautic/mautic/pull/12630#issuecomment-1805846995) for more context.

#### Steps to test this PR:

1. look at the contact list having a contact with a country set
2. depending on you error_reporting configuration of your server, you see no flags or you see an exception
![image](https://github.com/mautic/mautic/assets/556426/9b3c501c-4cdd-4ca6-adf1-c195a39c3b52)
3. patch
4. refresh list
5. see flags
![image](https://github.com/mautic/mautic/assets/556426/d16d9dfd-ccd5-4703-8c49-e5f13a5b7f2f)
